File: BubbleSort.java
public class BubbleSort {
    public static void bubbleSort(int[] A){
        int n = A.length;
        for(int i = 0 ; i < n- 1 ; i++){
            for(int j = 0 ; j < n - i - 1 ; j++){
                if(A[j] > A[j+1]){
                    int temp = A[j];
                    A[j] = A[j+1];
                    A[j+1] = temp;
                }
            }
        }
    }

    public static void printArray(int [] A){
        for(int i = 0 ; i < A.length ; i++){
            System.out.println(A[i] + " ");
        }
        System.out.println();
    }
    public static void main(String[] args) {

        int[] Arr = {4 , 6, 10,1 , 9 , 5};
        bubbleSort(Arr);
        printArray(Arr);


    }
}

File: BubbleSort.cpp
#include <bits/stdc++.h>
using namespace std;

void bubbleSort(int A[] , int size){
    for(int i = 0 ; i < size ; i++){
        for(int j = 0 ; j < size-i ; j++){
            if(A[j] > A[j+1]){
                //swapping
                int cpy = A[j];
                A[j] = A[j+1];
                A[j+1] = cpy;
            }
        }
    }

    for (int i = 0; i < size; i++)
    {
        cout << A[i] << " ";
    }
    
}

int main(){

    int Arr[7] = {4 , 1 , 7 , 9 , 3 , 2 , 8};
    bubbleSort(Arr , 7);
    cout << endl;

    return 0;
}
